modules/analysis-common/build.gradle

```diff
@@ -23,3 +24,12 @@ restResources {
 dependencies {
   compileOnly project(':modules:lang-painless')
 }
+
+tasks.named("yamlRestCompatTest").configure {
```

Comment:
One thing I realized recently is while the transform tasks are version-specific, this isn't. So when we bump the major we'll need to remove this if appropriate.

Reply:
I am thinking that maybe we want to make this source set and task name version specific (i.e. include v7 somewhere in there), like the transform task. I can follow up in a different PR. 

Hopefully there will very few black listed tests once all compatibility is applied, but your point still stands.